packages/node-fetch-server/src/lib/create-fetch-server.ts

```diff
@@ -0,0 +1,74 @@
+import * as http from 'node:http';
+
+import type { ClientAddress, ErrorHandler, FetchHandler } from './fetch-handler.js';
+
+import { FetchIncomingMessage as IncomingMessage } from './fetch-incoming-message/index.js';
+import { internalServerError } from './utils.js';
+import { sendResponse } from './request-listener.js';
+
+export interface RequestListenerOptions {
+  /**
+   * Overrides the host portion of the incoming request URL. By default the request URL host is
+   * derived from the HTTP `Host` header.
+   *
+   * For example, if you have a `$HOST` environment variable that contains the hostname of your
+   * server, you can use it to set the host of all incoming request URLs like so:
+   *
+   * ```ts
+   * createRequestListener(handler, { host: process.env.HOST })
+   * ```
+   */
+  host?: string;
+  /**
+   * An error handler that determines the response when the request handler throws an error. By
+   * default a 500 Internal Server Error response will be sent.
+   */
+  onError?: ErrorHandler;
+  /**
+   * Overrides the protocol of the incoming request URL. By default the request URL protocol is
+   * derived from the connection protocol. So e.g. when serving over HTTPS (using
+   * `https.createServer()`), the request URL will begin with `https:`.
+   */
+  protocol?: string;
+}
+
+export function createFetchServer(handler: FetchHandler, options?: RequestListenerOptions) {
+  let onError = options?.onError ?? defaultErrorHandler;
+
+  // @ts-expect-error
+  return http.createServer<IncomingMessage>(
+    { IncomingMessage },
```

Comment:
Interesting! I had no idea the `createServer` API could use a custom `IncomingMessage` class. Very cool find 👍